client/ConnectionThread.py

```diff
@@ -0,0 +1,109 @@
+import threading
+import socket
+import time
+
+
+class ConnectionThread(threading.Thread):
+    def __init__(self, thread_status):
+        threading.Thread.__init__(self)
+        self.handler_status = thread_status
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.conn = None
+        self.lock = threading.Lock()
+
+    def run(self):
+        while not self.handler_status.to_kill:
+            if not self.handler_status.is_connected:
+                if self.handler_status.is_active_connect:
+                    self.connect()
+                else:
+                    self.listen()
+            else:
+                break
+        self.conn.settimeout(3.0)
+        while not self.handler_status.to_kill:
+            self.send_counter()
+            time.sleep(1.0)
+            self.get_counter()
+            self.handler_status.msg += 1
+        self.kill()
+
+    def send_counter(self):
+        self.lock.acquire()
+        for i in range(5):
+            try:
+                msg = f"{self.handler_status.host_username} -> {self.handler_status.player_name}:{self.handler_status.msg}"
+                sent = self.conn.send(msg.encode())
+                print(f"Sent: {sent}")
+                break
+            except socket.error:
+                if i == 5:
+                    print("Couldn't send message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def get_counter(self):
+        self.lock.acquire()
+        for i in range(5):
+            try:
+                msg = self.conn.recv(2048)
+                msg = msg.decode()
+                if msg == '' or msg == ' ':
+                    print("Socket connection shut down")
+                    self.handler_status.is_connected = -1
+                print(msg)
+                break
+            except socket.error:
+                if i == 5:
+                    print("Couldn't receive message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def connect(self):
+        tries = 0
+        self.socket.settimeout(10.0)
+        print(f"Trying to connect with: {self.handler_status.player_ip}:{self.handler_status.player_port}")
+        while True:
+            try:
+                self.socket.connect((self.handler_status.player_ip, self.handler_status.player_port))
+                self.socket.settimeout(None)
+                self.conn = self.socket
```

Comment:
Cant get point of this operation

Reply:
After establishing connection, timeout should be reset. conn is used to differ listening socket from connection socket(they're different on listening client, but the same on actively connecting client)